multimedia/VLC/Portfile

```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
+# 1 viable solution as long as this is the case: drop the opencv filter.
+#     depends_lib-append  port:opencv
+#     # the +qt5 and/or +qt4 variants of port:opencv currently lead to build errors. A patch has
+#     # been submitted (but not included) that resolves the issue for qt4, not yet for qt5.
+#     # https://trac.macports.org/ticket/50709
+#     # For regular users a single check in the pre-fetch would suffice, but this maintainer
+#     # needs the check at other opportune moments too.
+#     proc check_opencv {} {
+#         global subport
+#         if {![active_variants opencv "" {qt4 qt5}]} {
+#             ui_error "${subport} cannot currently be built against port:opencv with the qt5 variant.\
+#             Install port:opencv without that variant and try again. This is only a build conflict; \
+#             after building you can reactivate your preferred opencv variant"
+#             return -code error "Install or activate port:opencv without the qt5 variant before building ${subport}"
+#         }
+#     }
+#     pre-fetch {
+#         check_opencv
+#     }
+#     pre-configure {
+#         check_opencv
+#     }
+#     pre-build {
+#         check_opencv
+#     }
+#     patchfiles-append   patch-opencv_includes.diff
+#     # The opencv example apparently builds on 10.11 but I'm having issues with it on 10.9
+#     # it doesn't appear to be indispensable, so simply skip it.
+#     patchfiles-append   patch-no-opencv_example.diff
+
+    # VLC 2.2.2 should have full Lua 5.3 compatibility, for now, depend on port:lua52
+    # That requires the patch-for-lua52.diff which is appended to patchfiles below.
+    # update 2.2.2 : lua 5.3 gives compiler syntax errors so we stick with 5.2 for now.
+    depends_lib-append  port:lua52
+    # depends_lib-append  port:lua
+
+    # VLC doesn't currently build for me using port:clang-3.8
+    compiler.blacklist-append \
+                        gcc-4.2 llvm-gcc-4.2 macports-clang-3.8 {clang < 300}
+
+    patchfiles          buildfix-package.mak.patch \
+                        configure.ac-no-arch.patch \
+                        PR-34741-no__clang_version__.patch \
+                        static_assert.patch \
+                        no-sparkle.patch \
+                        patch-vlc-no-O4.diff \
+                        patch-ffmpegVLC.diff
 
-compiler.blacklist  gcc-4.2 llvm-gcc-4.2 {clang < 300}
+    post-patch {
+        reinplace "s:librsvg-2/librsvg:librsvg:" \
+            ${worksrcpath}/modules/text_renderer/svg.c
 
-pre-fetch {
-    if {${os.platform} eq "darwin" && ${os.major} < 10} {
-        ui_error "${name} ${version} requires Mac OS X 10.6 or greater."
-        return -code error "incompatible Mac OS X version"
+        if {[string match *clang* ${configure.cxx}] && ${configure.cxx_stdlib} == "libc++"} {
+            reinplace "s:-lstdc\+\+:-lc++:" \
+                ${worksrcpath}/configure.ac \
+                ${worksrcpath}/modules/access/Makefile.am
+        }
```

Comment:
That's not mine, it's been in since at least v2.1.5 .
The replacement expression is safe and as far as I'm concerned it's exactly the sort of operation for which you'd use reinplace.